### PR TITLE
docs: sphinx: Document potential for header and library conflicts

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
https://trello.com/c/8SQBNzbJ/39-document-potential-build-failure-causes

Document the potential for conflicts between system-provided and superbuild-provided packages.

Testing: Check docs

### CODE REVIEW

**Comment by mtbc** [comment]
Where are the docs? http://www.openmicroscopy.org/site/support/contributing/ci-docs.html gets me to places like http://www.openmicroscopy.org/site/support/ome-model-staging/ but I remain a bit lost I'm afraid.

**Comment by sbesson** [comment]
@mtbc: see http://www.openmicroscopy.org/site/support/contributing/ci-ome-files.html#term-ome-files-cpp-dev-merge-docs. We might want to review how we organize our jobs by docs vs component /cc @hflynn

**Comment by hflynn** [comment]
https://www.openmicroscopy.org/site/support/ome-files-cpp-staging/ome-cmake-superbuild/manual/html/index.html @mtbc

**Comment by hflynn** [comment]
Ah yes, I'll open a PR to add these jobs to the docs page.

File: docs/sphinx/overview.rst [hunk 1]
```diff
@@ -17,11 +17,20 @@ the same compiler options, to ensure that all the libraries and
...
-of the third-party dependencies provided by the package manager.  This
-is supported.  However, when building for older distributions with
-outdated dependencies, it can be used to create a self-contained set
-of newer libraries which won't conflict with the base platform.
+comprehensive set of packages, it may be desirable to disable building
+of third-party dependencies which are already provided by the package
+manager.  This is supported.  When building for older distributions,
+where the base platform has outdated or missing dependencies, the
+super-build can be used to create a self-contained set of newer
+libraries which won't conflict with the base platform.  However, if a
```
Comment (mtbc) [inline docs/sphinx/overview.rst]: will not

File: docs/sphinx/overview.rst [hunk 2]
```diff
@@ -17,11 +17,20 @@ the same compiler options, to ensure that all the libraries and
...
+package both exists on the system and is built by the super-build, it
+could potentially result in a mismatching set of headers and libraries
```
Comment (mtbc) [inline docs/sphinx/overview.rst]: not sure "potentially" adds anything to "could"?

**Comment by rleigh-codelibre** [comment]
Fixes pushed.

**Comment by sbesson** [comment]
@mtbc: are you fine with the current state of this PR?

**Comment by mtbc** [comment]
Will relist it for today if it's not already there.

**Review by mtbc — APPROVED** [review]

**Pull request merged**